feat(rbac): Allow for custom RBAC policies to be configured with ket

jordanp-atlnz opened the pull request:
This patch allows for custom rbac file to be provided to configure extra
rbac rules for the test runner pod. This will allow for things like
OpenFeature to have RBAC policies honored by the testing system.

Implementation is as follows:

- Allow a new option `--rbac` to pass a file containing RBAC rules to KET.
- The `--rbac` file gets merged into the existing default RBAC rules.
- Integration test updated to encompass new behaviour.

jordanp-atlnz committed:
feat(rbac): Allow for custom RBAC policies to be configured with ket

This patch allows for custom rbac file to be provided to configure extra
rbac rules for the test runner pod. This will allow for things like
OpenFeature to have RBAC policies honored by the testing system.

spmiller reviewed (changes requested):
Looks good. A minor nit to investigate.

jordanp-atlnz committed:
chore(rbac): omit the handling of 0 length additional rules

Remove the handling of length 0 rules as it is not needed; the function
already supports merging of rules that are `nil`.

spmiller reviewed (approved):
Looks good
